docs/dev/architecture.md

```diff
@@ -33,11 +33,43 @@ Coming soon&hellip;
 
 ## 3. Code Organization
 
-Coming soon&hellip;
+The worker agent codebase is organized into several key directories:
+
+```
+deadline-cloud-worker-agent/
+├── src/
+│   └── deadline_worker_agent/  # Main source code
+│       ├── aws/                # AWS service APIs
+│       │   └── deadline/       # AWS Deadline Cloud APIs
+│       ├── aws_credentials/    # AWS credentials management
+│       ├── boto/               # Boto3 and botocore configuration and shim layer
+│       ├── config/             # Configuration handling
+│       ├── installer/          # Worker agent installation logic
+│       ├── linux/              # Linux-specific code
+│       ├── log_sync/           # Log synchronization
+│       ├── scheduler/          # Worker scheduler
+│       ├── sessions/           # Session management
+│       │   ├── actions/        # Session actions
+│       │   │   └── scripts/    # Helper scripts for actions
+│       │   └── job_entities/   # Job entity handling
+│       ├── startup/            # Startup and shutdown phase
+│       └── windows/            # Windows-specific code
+├── pipeline/                   # CI/CD pipeline scripts
+├── scripts/                    # Utility scripts for development and testing
+├── tests/
+│   ├── e2e/                    # End-to-end tests
+│   ├── integration/            # Integration tests
+│   └── unit/                   # Unit tests
+└── docs/                       # Documentation
```

Comment:
It would be nice if we could mechanize keeping this up to date, but there's not likely to be too many organizational changes that i think it should be fine.